### DISCUSSION TITLE
How can i know which value to put in a setter for a `duration` type field?

### DISCUSSION THREAD

**Discussion by D4yvid** [slint-ui/slint#6372] [Q&A] [answered]
I created a global which there's a field called `animation-duration`, that is of the type `duration`:
```slint
export global Animator {

    in-out property <duration> animation-duration: 120ms;
    in-out property <duration> ripple-animation-duration: 2s;

}
```

This property has a setter, that receives a `i64` in the parameter, how can i know what to put in here? Should i use milliseconds, seconds, use what? Why not using a std::time::Duration as the argument instead of i64?

Here's the code:
```rust
    // Animator::set_animation_duration(value: i64)
    animator.set_animation_duration(120); // 120 milliseconds? 120 seconds? 120 what?
```

**Comment by ogoffart** [comment, marked as answer]
Yes, this is milliseconds. Documented there: https://releases.slint.dev/1.8.0/docs/rust/slint/docs/type_mappings/

> Why not using a std::time::Duration as the argument instead of i64?

That is a very good question. The reason is that we can't use `std::time::Duration` as field of a generated struct or property because it is not `repr(C)` and because we also support other language like C++, it needs to be binary compatible.

We could have made so that we use `std::time::Duration` for things that are only exposed to rust, and u64 for things that needs to be shared, but unfortunately we didn't do that before Slint 1.0. (Would have made things more complicated)

**Reply by D4yvid** [reply]
Thanks for the reply! Now i understand why cannot use std::time::Duration instead!
